src/LiveDevelopment/Agents/RemoteFunctions.js

```diff
@@ -250,17 +269,149 @@ function RemoteFunctions(experimental, remoteWSPort) {
             }
             return false;
         },
-
         _makeHighlightDiv: function (element, doAnimation) {
             var elementBounds = element.getBoundingClientRect(),
                 highlight = window.document.createElement("div"),
-                styles = window.getComputedStyle(element);
+                elementStyling = window.getComputedStyle(element),
+                transitionDuration = parseFloat(elementStyling.getPropertyValue('transition-duration')),
+                animationDuration = parseFloat(elementStyling.getPropertyValue('animation-duration'));
+			
+            if (transitionDuration) {
+                animateHighlight(transitionDuration);
+            }
+
+            if (animationDuration) {
+                animateHighlight(animationDuration);
+            }
 
             // Don't highlight elements with 0 width & height
             if (elementBounds.width === 0 && elementBounds.height === 0) {
                 return;
             }
+            
+            var realElBorder = {
+              "right": elementStyling.getPropertyValue('border-right-width'),
+              "left": elementStyling.getPropertyValue('border-left-width'),
+              "top": elementStyling.getPropertyValue('border-top-width'),
+              "bottom": elementStyling.getPropertyValue('border-bottom-width'),
+            };
+            
+            var innerWidth = elementBounds.width - parseInt(realElBorder.left) - parseInt(realElBorder.right);
+            var innerHeight = elementBounds.height - parseInt(realElBorder.top) - parseInt(realElBorder.bottom);
+          
+            var visualisations = [];
```

Comment:
This should be an object literal instead of assigning `horizontal` and `vertical` as array properties?

``` js
var visualisations = {
 horizontal: "left, right",
 vertical: "top, bottom",
}
```